### PR TITLE
feat(dev/linkTask): Linker module implemented to link two tasks

### PR DESCRIPTION
**Pull request by printf-twinkle**
### Functionality:
The link module is used to create links between two tasks- parent and child task. This feature is helpful when a task has to be divided among people or when a task has some dependencies like dev team depending on the ui team for designs.

### Solution:
The parentTaskId and childTaskId are used to create two links so that their tasks can be connected.

### Risk level:
- [x] high
- [ ] medium
- [ ] low

### How to test:
Go to a task, click on the three dots and go to create a subTask, fill in all details and after creating the task correctly, a new subTask will be created from the parentTask and their links will be set accordingly.

### CODE REVIEW

File: src/app/body/task-details/sub-task/sub-task.component.ts [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+  }
+
+  close() {
+    jQuery('#createNewTask').modal('hide');
+    jQuery('#form').trigger("reset");
+    this.taskCreated.emit({ completed: true });
+  }
+
+}
+
```
Comment (viv9k) [inline src/app/body/task-details/sub-task/sub-task.component.ts:167]: Remove subtask component

File: src/app/body/task-details/task-details.component.html [hunk 2]
```diff
@@ -287,5 +291,9 @@ <h4 id="value">{{task.Title}}</h4>
     <ng-container *ngIf="linkEnabled" id="linkModalLabel">
         <app-link [taskId]="Id" [orgDomain]="orgDomain" (addedLink)="addedLink($event)"></app-link>
     </ng-container>
+    <!-- <ng-container *ngIf="subtaskEnabled" id="exampleModalCenterLabel">
+        <app--new-task [Id]="task.Id" ></app-create-new-task>
+        <app-sub-task [task]="task" (addedSubtask)=" addedSubtask($event)"></app-sub-task> 
+    </ng-container> -->
```
Comment (viv9k) [inline src/app/body/task-details/task-details.component.html:297]: Remove commented code

File: src/app/body/task-details/task-details.component.ts [hunk 3]
```diff
@@ -31,6 +33,8 @@ export class TaskDetailsComponent implements OnInit {
   editTaskEnabled: boolean = false
   deleteTaskEnabled: boolean = false
   linkEnabled: boolean = false
+  createNewTaskEnabled: boolean = false;
+  // subtaskEnabled: boolean = false;
```
Comment (viv9k) [inline src/app/body/task-details/task-details.component.ts:37]: Remove commented code

File: src/app/body/task-details/task-details.component.ts [hunk 4]
```diff
@@ -31,6 +33,8 @@ export class TaskDetailsComponent implements OnInit {
   editTaskEnabled: boolean = false
   deleteTaskEnabled: boolean = false
   linkEnabled: boolean = false
+  createNewTaskEnabled: boolean = false;
```
Comment (viv9k) [inline src/app/body/task-details/task-details.component.ts:36]: This variable is not required

File: src/app/body/task-details/sub-task/sub-task.component.spec.ts [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+  beforeEach(() => {
+    fixture = TestBed.createComponent(SubTaskComponent);
+    component = fixture.componentInstance;
+    fixture.detectChanges();
+  });
+
+  it('should create', () => {
+    expect(component).toBeTruthy();
+  });
+});
```
Comment (viv9k) [inline src/app/body/task-details/sub-task/sub-task.component.spec.ts:25]: Remove subtask component

File: src/app/body/create-new-task/create-new-task.component.ts [hunk 6]
```diff
@@ -143,21 +149,53 @@ export class CreateNewTaskComponent implements OnInit {
...
+        const orgDomain = this.backendService.getOrganizationDomain();
+        const linkURL = "http://127.0.0.1:4200/TaskDetails/" + this.parentTaskId;
+        await callable({OrgDomain: orgDomain, TaskID: this.childTaskId, LinkType: "child", linkURL: linkURL}).toPromise();
+      }
+      catch {
+
+      }
+    }
+  }
+
```
Comment (viv9k) [inline src/app/body/create-new-task/create-new-task.component.ts:198]: Revert this back to normal

File: functions/model/linker/tark/setLinkDetails.js [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+    let status = 200;
+    let result;
+
+    const promise = getTask(taskId, orgDomain).then((taskDetail) => {
+        if (taskDetail == undefined) {
+            result = {data: {status: "ERROR"}};
+        } else {
+            const linkCounter = taskDetail.LinkCounter + 1;
+            const linkId= "Link"+(linkCounter);
+            console.log("hereeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee")
```
Comment (viv9k) [inline functions/model/linker/tark/setLinkDetails.js:30]: remove unnecessary console.log

**Review by viv9k — CHANGES_REQUESTED** [review]
Please fix the required changes

File: src/app/body/task-details/sub-task/sub-task.component.ts (hunk 1 above)
Comment (printf-twinkle) [reply src/app/body/task-details/sub-task/sub-task.component.ts:167]: fixed

File: src/app/body/task-details/task-details.component.html (hunk 2 above)
Comment (printf-twinkle) [reply src/app/body/task-details/task-details.component.html:297]: removed

File: src/app/body/task-details/task-details.component.ts (hunk 3 above)
Comment (printf-twinkle) [reply src/app/body/task-details/task-details.component.ts:37]: done

File: src/app/body/task-details/task-details.component.ts (hunk 4 above)
Comment (printf-twinkle) [reply src/app/body/task-details/task-details.component.ts:36]: removed

File: src/app/body/task-details/sub-task/sub-task.component.spec.ts (hunk 5 above)
Comment (printf-twinkle) [reply src/app/body/task-details/sub-task/sub-task.component.spec.ts:25]: done

File: functions/model/link/lib.js [hunk 8]
```diff
@@ -0,0 +1,25 @@
+/* eslint-disable linebreak-style */
+/* eslint-disable no-undef */
+/* eslint-disable object-curly-spacing */
+/* eslint-disable eol-last */
+/* eslint-disable indent */
+/* eslint-disable max-len */
+
+const { db } = require("../application/lib");
+
+
```
Comment (viv9k) [inline functions/model/link/lib.js:10]: Remove model link

**Review by viv9k — CHANGES_REQUESTED** [review]
Remove model link

File: functions/model/linker/tark/setLinkDetails.js (hunk 7 above)
Comment (printf-twinkle) [reply functions/model/linker/tark/setLinkDetails.js:30]: done

File: functions/model/link/lib.js (hunk 8 above)
Comment (printf-twinkle) [reply functions/model/link/lib.js:10]: done

File: src/app/body/task-details/task-details.component.html (hunk 2 above)
Comment (viv9k) [reply src/app/body/task-details/task-details.component.html:297]: Here

File: src/app/body/create-new-task/create-new-task.component.ts [hunk 9]
```diff
@@ -1,4 +1,4 @@
-import { Component, EventEmitter, OnInit, Output, ViewChild } from '@angular/core';
+import { Component, EventEmitter, Input, OnInit, Output, ViewChild } from '@angular/core';
```
Comment (viv9k) [inline src/app/body/create-new-task/create-new-task.component.ts:1]: revert this

File: src/app/body/task-details/task-details.component.ts [hunk 10]
```diff
@@ -203,6 +200,10 @@ export class TaskDetailsComponent implements OnInit {
     this.linkEnabled = false;
   }
 
+  // addedSubtask( data: { completed: boolean } ) {
+  //   this.subtaskEnabled = false;
+  // }
+
```
Comment (viv9k) [inline src/app/body/task-details/task-details.component.ts:206]: remove commented code

**Review by viv9k — CHANGES_REQUESTED** [review]
Please change it

File: functions/model/linker/tark/setLinkDetails.js [hunk 11]
```diff
@@ -6,7 +6,9 @@
 /* eslint-disable max-len */
 // eslint-disable-next-line no-dupe-else-if
 
-const { getTask, setLinkDoc, updateTask } = require("../lib");
+const { getTask } = require("../../tasks/lib");
+const { setLinkDoc } = require("../lib")
+const { updateTask } = require("../../tasks/lib");
```
Comment (viv9k) [inline functions/model/linker/tark/setLinkDetails.js:11]: Here getTask and updateTask can be same require import

**Review by viv9k — CHANGES_REQUESTED** [review]
Please do the required fix

**Comment by viv9k** [comment]
change target branch

File: src/app/body/create-new-task/create-new-task.component.css [hunk 12]
```diff
@@ -46,7 +35,6 @@
     border-radius: 6px;
     opacity: 1;
     margin: 5px !important;
-    position: fixed;
```
Comment (viv9k) [inline src/app/body/create-new-task/create-new-task.component.css:49]: Why changing the css?

File: src/app/body/create-new-task/create-new-task.component.html [hunk 13]
```diff
@@ -4,22 +4,20 @@
...
                     <h2 class="modal-title " id="createNewTaskTitle"><strong>Create New Task</strong></h2>
-                    <button type="button" class="close" (click)= "close()" aria-label="Close">
+                    <button type="button" class="close" (click)="close()" aria-label="Close">
                         <span aria-hidden="true" class="material-icons">close</span>
                     </button>
                 </div>
-                <div class="modal-body container-fluid pt-0 mt-n4">
+                <div class="modal-body container-fluid pt-0 mt-4">
                     <div class="medium ml-1">{{ todayDate }}</div>
-                    <br>
```
Comment (viv9k) [inline src/app/body/create-new-task/create-new-task.component.html:13]: Revert the changes in this file.

**Review by viv9k — CHANGES_REQUESTED** [review]
Revert unnecessary changes

**Comment by viv9k** [comment]
Resolve merge conflict properly

**Review by viv9k — APPROVED** [review]
Approved
